curp/src/server/raw_curp/state.rs

```diff
@@ -221,29 +222,28 @@ impl MajorityConfig {
         }
     }
 
-    /// Get voters set
+    /// Get voters of current config
     pub(super) fn voters(&self) -> &HashSet<ServerId> {
         &self.voters
     }
 
-    /// Insert a voter
-    pub(super) fn insert(&mut self, id: ServerId) -> bool {
-        self.voters.insert(id)
+    /// Get quorum: the smallest number of servers who must be online for the cluster to work
+    fn quorum(&self) -> usize {
+        self.voters.len() / 2 + 1
     }
+}
 
-    /// Remove a voter
-    pub(super) fn remove(&mut self, id: ServerId) -> bool {
-        self.voters.remove(&id)
-    }
+impl Deref for MajorityConfig {
```

Comment:
Why do we need Deref here? I believe it should only be used for smart pointers.

Reply:
Simplified operation, `cfg.voters.insert(id)` -> `cfg.insert(id)`

Reply:
See https://doc.rust-lang.org/std/ops/trait.Deref.html

> Implementing Deref for smart pointers makes accessing the data behind them convenient, which is why they implement Deref. On the other hand, the rules regarding Deref and [DerefMut](https://doc.rust-lang.org/std/ops/trait.DerefMut.html) were designed specifically to accommodate smart pointers. Because of this, Deref should only be implemented for smart pointers to avoid confusion.

Reply:
Fixed